server/routes/orderAPI.js

```diff
@@ -0,0 +1,132 @@
+  
+
+// =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+  function idGenerator() {
+    function s4() {
+      return Math.floor((1 + Math.random()) * 0x10000).toString(16).substring(1);
+    }
+    return s4() + s4() + '-' + s4() + '-' + s4() + '-' + s4() + '-' + s4() + s4() + s4();
+  }
+
+// =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+  router.get('/', function( req, res) {
+
+    // simple instructions on API paths
+    // access point: http://localhost:3000/orderAPI
+    res.send('<b>Order API Documentation</b>' +
+      '<br><br>List Database: http://localhost:3000/orderAPI/database' +
+      '<br><br>Post Search Request: http://localhost:3000/orderAPI/search' +
+      '<br><br>Post Order Placement Request: http://localhost:3000/orderAPI/placement'
+      );
+  });
+
+
+// =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+  router.get('/database', function(req, res) {
+
+    // display temporary database
+    // access point: http://localhost:3000/orderAPI/database
+    res.json(orderDatabase);
+  });
+
+
+// =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+  router.post('/search', function(req, res) {
+
+    // find order details
+    // http://localhost:3000/orderAPI/search
+    var orderDetails = {};
+    var serverMessage = '';
+    var searchRequest = req.body;
+
+    // simulate search send {"orderId" : 1}
+    // Triggered server response when order is placed
+    console.log('Searching for order ' + searchRequest.orderId);
+
+    if(orderDatabase.length <= searchRequest.orderId){
+      serverMessage = 'orderId: ' + searchRequest.orderId + ' was not found'
+    }else{
+      orderDetails  = orderDatabase[searchRequest.orderId];
+      serverMessage = 'orderId: ' + searchRequest.orderId + ' located'
+    }
+
+
+    res.json({
+      serverResponse : 'Searching for order: ' + searchRequest.orderId,
+      serverMessage : serverMessage,
+      orderDetails: orderDetails
+    });
+  });
+
+
+// =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+//This is what we expect an order json to look like
+  var exampleOrder =
+  {"order" :
+  {"recipient" :
+  {"name" : "Jazy Jasilo",
+    "address" : "3300 University Blvd, Winter Park, FL 32792",
+    "email" : "orange@fullsail.edu",
+    "phone" : "555-555-5555"},
+    "units" : [{
+      "unitId" : "a5296ab9-9eee-7ba0-0a79-b801594f2c91",
+      "quantity" : 4}
+    ]}
+  };
+
+// =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+router.post('/placement', function(req, res) {
+  var orderPlacement = req.body;
+  orderTracking = { status : 'packing', timestamp : 'today'};
+  orderRecipient = orderPlacement.order.recipient;
+  orderUnits = orderPlacement.order.units;
+
+
+  var placement = {};
+  placement.tracking = orderTracking;
+  placement.recipient = orderRecipient;
+  placement.unit = orderUnits;
```

Comment:
Placement isn't a very good definition for what is being done here, maybe try thinking of another variable name.